activiti-core/activiti-api-impl/activiti-api-process-runtime-impl/src/main/java/org/activiti/runtime/api/impl/ProcessVariablesPayloadValidator.java

```diff
@@ -63,7 +63,7 @@ public ProcessVariablesPayloadValidator(DateFormatterProvider dateFormatterProvi
                                                       null;
 
         return Optional.ofNullable(processExtensionModel)
-                .map(ProcessExtensionModel::getExtensions)
+                .map(p -> p.getExtensionsByProcessDefinitionId(processDefinitionId))
```

Comment:
I thought we did not have the processDefinitionId in the extensions json
I thought we had only the process definition key in the extensions json

Reply:
processDefinitionId is basically the combination of processDefinitionKey:appVersion:deploymentId and it's immutable as far as i know in the whole life of that version of the process.